gtn/hd_span.h

```diff
@@ -151,6 +184,16 @@ class HDSpan {
     data_ = newData;
   };
 
+  void reserve(size_t size) {
+    if (size <= capacity_) {
+      return;
+    }
+    capacity_ = size;
+    auto newData = allocAndCopy(data_);
+    free();
```

Comment:
shouldn't we set size/capacity to zero in the call to `free()`. Then L191 would be of no effect.

Reply:
Its a little complicated at the moment :\. The free is really just a convenience function for freeing the memory which first checks the device. The clear function actually sets the size to zero and frees the pointer. I could change "free" to "delete" or something like that if it would be less confusing?